runtime/src/bank.rs

```diff
@@ -1637,6 +1689,297 @@ impl Bank {
             .fetch_add(collected_rent, Ordering::Relaxed);
     }
 
+    fn collect_rent_eagerly(&self) {
+        if !self.enable_eager_rent_collection() {
+            return;
+        }
+
+        let mut measure = Measure::start("collect_rent_eagerly-ms");
+        for partition in self.rent_collection_partitions() {
+            self.collect_rent_in_partition(partition);
+        }
+        measure.stop();
+        inc_new_counter_info!("collect_rent_eagerly-ms", measure.as_ms() as usize);
+    }
+
+    fn enable_eager_rent_collection(&self) -> bool {
+        if self.lazy_rent_collection.load(Ordering::Relaxed) {
+            return false;
+        }
+
+        let not_yet_for_existing_clusters = (self.operating_mode() == OperatingMode::Stable
+            && self.genesis_hash.unwrap() == MAINNET_BETA_GENESIS_HASH
```

Comment:
We don't need this check.  Assuming that `OperatingMode::Stable` is mainnet-beta, and `OperatingMode::Preview` will be fine.

Reply:
I see! I just reverted this code from this pr. I'll take similar gating approach you took with #9815 / #9837  :)